.github/workflows/multi-arch-test-build.yml

```diff
@@ -35,9 +35,17 @@ jobs:
       - name: Determine changed packages
         run: |
           # only detect packages with changed Makefiles
-          PACKAGES="$(git diff --diff-filter=d --name-only origin/master \
-            | grep -E 'Makefile$|test.sh$' | grep -Ev '/files/|/src/' \
-            | awk -F/ '{ print $(NF-1) }' | tr '\n' ' ')"
+          PKG_ROOTS=$(find -name Makefile | sed -e 's@./\(.*\)/Makefile@\1/@')
+          CHANGES=$(git diff --diff-filter=d --name-only origin/master)
+
+          for ROOT in $PKG_ROOTS; do
+            for CHANGE in $CHANGES; do
+              if [[ "$CHANGE" == "$ROOT"* ]]; then
+                PACKAGES+=$(echo $ROOT | sed -e 's@.*/\(.*\)/@\1 @')
```

Comment:
Doesn't this adds duplicates if two changes happened in a single package.

Reply:
yeah, the `break` below kind of fixes that, by moving onto the next ROOT as soon as a match is found, so duplicates should not match again.

Reply:
Did you run a benchmark on how long this loop actually takes?

Reply:
I think it's negligible - I saw CI kick off and like 10 seconds later it was already past that step. On my machine it takes about 200ms to run the script, but CI is likely slower. Let me add some debug in my temp commit.

Reply:
Okay nevermind then, thanks for checking